crimemapper/views.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+"""View functions created here."""
+from pyramid.view import view_config
+from crimemapper.models import (
+    DBSession,
+    Entry,
+)
+import os
+from .crimedict import CRIME_DICT
+from pyramid.httpexceptions import HTTPServiceUnavailable
+from .graph_calcs import crime_dict_totals, crime_category_breakdown
+
+CACHED_RESULTS = {}
+
+
+def cached_db_call():
+    """Cache database hit so no further calls need to be made."""
+    if 'already_called' not in CACHED_RESULTS:
+        results = DBSession().query(
+            Entry.latitude,
+            Entry.longitude,
+            Entry.summarized_offense_description
+        ).all()
+        CACHED_RESULTS['already_called'] = results
+    return CACHED_RESULTS['already_called']
+
+
+def find_category(description):
+    """Return category of crime in dictionary."""
+    for k, v in CRIME_DICT.items():
+        if description in v:
+            return k
+
+
+@view_config(route_name='map', renderer='templates/map.jinja2')
+def map_view(request):
+    """Render map view on page."""
+    point = cached_db_call()
+    places = []
+    for i, l in enumerate(point):
```

Comment:
not particularly pythonic to use `i` here like you are.  Would be better this way:

```python
points = cached_db_call()
for point in points:
    place = {'lat': point[0], 'lng': point[2]}
    description = str(point[2])
    category = find_category(description)
    places.append([place, description, category])
````